addons/project/static/src/components/notebook_task_one2many_field/notebook_task_list_renderer.js

```diff
@@ -0,0 +1,37 @@
+/** @odoo-module **/
+
+import { ListRenderer } from '@web/views/list/list_renderer';
+
+import { useState } from "@odoo/owl";
+
+export class NotebookTaskListRenderer extends ListRenderer {
+    static rowsTemplate = "project.NotebookTaskListRenderer.Rows";
+    setup() {
+        super.setup();
+        this.hideState = useState({
+            hide: localStorage.getItem(this._getStorageKey) == 'true' || false,
+        });
+    }
+
+    /**
+     * @private
+     * @returns {string}
+     */
+    get _getStorageKey() {
+        return `hide_closed_${this.constructor.name}`;
+    }
+
+    get hideClosed() {
+        return this.hideState.hide
+    }
+
+    get ShowX2MRecords() {
+        return this.props.list.context.closed_X2M_count > 0 || typeof this.props.list.context.closed_X2M_count === 'undefined';
```

Comment:
why are we showing the buttons when the second condition is met again ? kinda confused with this :thinking:

Reply:
Mmh yeah i guessed that it should always be possible to display the button even if the current x2m task field doesn't have any "closed_X2M_count" fields that could track the number of closed task in this x2m dynamically.

For exemple in this PR, i needed to add a closed_depend_on_count computed field for the blocked by task field count, i don't think this should be necessary for every potential x2m task field displayed in a tree view

So if i don't specify the closed_X2M_count we always display the Show/Hide button
Tell me if that makes sense for you

Reply:
TLDR: basically if the user does not set up this prop in the view we always display the "show closed tasks" button
Alternativelly we could do that count in the frontend and not use a counter field at all, but since it's already there for the subtasks i thought it would be ok to create another computed field.
![image](https://github.com/odoo/odoo/assets/36259462/421e50e3-8267-46b3-8716-ddab1096c4bd)